src/test/java/com/connexta/store/StoreITests.java

```diff
@@ -6,541 +6,388 @@
+
+  private static final String TEST_FILE_CONTENT = "some-content";
+  private static final Resource TEST_FILE_RESOURCE =
+      new ByteArrayResource(TEST_FILE_CONTENT.getBytes()) {
+
+        @Override
+        public long contentLength() {
+          return TEST_FILE_CONTENT.getBytes().length;
+        }
+
+        @Override
+        public String getFilename() {
+          return ORIGINAL_FILENAME;
+        }
+      };
+  private static final String TEST_METACARD_CONTENT = "metacard-content";
+  private static final Resource TEST_METACARD_RESOURCE =
+      new ByteArrayResource(TEST_METACARD_CONTENT.getBytes()) {
+
+        @Override
+        public long contentLength() {
+          return TEST_METACARD_CONTENT.getBytes().length;
+        }
+
+        @Override
+        public String getFilename() {
+          return "metacard.xml";
+        }
+      };
+
+  @Container
+  private static final GenericContainer s3Container =
+      new GenericContainer("localstack/localstack:0.10.5")
+          .withExposedPorts(LOCALSTACK_PORT)
+          .withEnv("SERVICES", "s3")
+          .waitingFor(new LogMessageWaitStrategy().withRegEx(".*Ready\\.\n"));
+
+  @TestConfiguration
+  static class Config {
+
+    @Bean
+    public AmazonS3Configuration testAmazonS3Configuration() {
+      return new AmazonS3Configuration(
+          String.format(
+              "http://%s:%d",
+              s3Container.getContainerIpAddress(), s3Container.getMappedPort(LOCALSTACK_PORT)),
+          "local",
+          MINIO_ADMIN_ACCESS_KEY,
+          MINIO_ADMIN_SECRET_KEY);
+    }
+  }
+
+  private MockRestServiceServer indexMockRestServiceServer;
+
+  @Inject private AmazonS3 amazonS3;
+
+  @Inject
+  @Named("nonBufferingRestTemplate")
+  private RestTemplate nonBufferingRestTemplate;
+
+  @Inject
+  @Named(TransformConfiguration.TRANSFORM_CLIENT_BEAN)
+  private WebClient transformWebClient;
+
+  @Value("${endpoints.store.version}")
+  private String storeApiVersion;
+
+  @Value("${endpoints.store.url}")
+  private String storeUrl;
+
+  @Value("${endpoints.transform.url}")
+  private String transformUrl;
+
+  @Value("${endpoints.ingest.version}")
+  private String ingestVersion;
+
+  @Value("${s3.bucket.file}")
+  private String fileBucket;
+
+  @Value("${s3.bucket.irm}")
+  private String irmBucket;
+
+  @Value("${s3.bucket.metacard}")
+  private String metacardBucket;
+
+  private static final MultiValueMap<String, HttpEntity<?>> TEST_INGEST_REQUEST_BODY;
+
+  static {
+    final MultipartBodyBuilder builder = new MultipartBodyBuilder();
+    builder.part("file", TEST_FILE_RESOURCE);
+    builder.part("metacard", TEST_METACARD_RESOURCE);
+    builder.part("correlationId", "000f4e4a");
+    TEST_INGEST_REQUEST_BODY = builder.build();
+  }
+
+  private MockWebServer mockTransformServer;
+  private MockWebServer mockFileServer;
+
+  @BeforeEach
+  void beforeEach() throws Exception {
+    mockTransformServer = new MockWebServer();
+    mockTransformServer.start(9090);
```

Comment:
✏️ This had to match the port of the transform service passed in via the test spring config in order for the mocker server to properly return the mock responses. I couldn't figure out how to start the mock server on a random port, then convey that to spring as the transform service's port. Any ideas?